src/main/scala/battleship/GameActor.scala

```diff
@@ -0,0 +1,111 @@
+package battleship
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import battleship.GameActor._
+import battleship.PlayerActor.{GetNextShot, PlaceBoats}
+import battleship.game._
+
+/**
+  * Created by m06f947 on 3-8-2017.
+  */
+
+
+
+object GameActor {
+
+  case class StartGame(boardsize:Int, player1:ActorRef, player2:ActorRef, boatset: Seq[Boat])
+
+  //case class WhatIsBoatSetup(boats:Seq[Boat], boardsize: Int)
+  case class ThisIsBoatSetup(placement: Set[(Boat,BoatLocation)])
+  //case class WhatIsNextMove(boardsize:Int, shotHistory: Seq[((Int,Int),ShotResult)])
+  case class ThisIsNextMove(x: Int, y: Int)
+
+
+  def props: Props = Props(new GameActor)
+
+}
+
+class GameActor extends Actor with ActorLogging {
+
+
+  //internal state
+  var boardsize = Game.defaultBoardSize
+  var boatSet = Game.defaultBoatSet
+  //first of pair is currentplayer
+  var players: (ActorRef, ActorRef) = (Actor.noSender, Actor.noSender)
+  def currentPlayer = players._1
+  def swapPlayers = {players = (players._2, players._1)}
+  //Have the boatsetup ready. In here player1 is boatsetup._1

```

Comment:
Second part of this comment is incorrect. `boatsetup` is not a tuple so `boatsetup._1` is not a valid expression.